packages/arb-util/broadcaster/confirmedaccumulatorcatchupbuffer.go

```diff
@@ -41,23 +41,41 @@ func (q *ConfirmedAccumulatorCatchupBuffer) OnRegisterClient(ctx context.Context
 	// Ignore messages older than requested sequence number
 	if clientConnection.RequestedSeqNum().Cmp(big.NewInt(0)) > 0 {
 		requestedLastSeqNum := new(big.Int).Sub(clientConnection.RequestedSeqNum(), big.NewInt(1))
-		for ; startingIndex < len(q.broadcastMessages); startingIndex++ {
-			if q.broadcastMessages[startingIndex].FeedItem.BatchItem.LastSeqNum.Cmp(requestedLastSeqNum) >= 0 {
-				break
+		if q.broadcastMessages[0].FeedItem.BatchItem.LastSeqNum.Cmp(requestedLastSeqNum) < 0 {
+			startingIndex = int(new(big.Int).Sub(requestedLastSeqNum, q.broadcastMessages[0].FeedItem.BatchItem.LastSeqNum).Uint64())
+			comparison := q.broadcastMessages[startingIndex].FeedItem.BatchItem.LastSeqNum.Cmp(requestedLastSeqNum)
+			if comparison < 0 {
+				startingIndex++
+				for ; startingIndex < len(q.broadcastMessages); startingIndex++ {
+					if q.broadcastMessages[startingIndex].FeedItem.BatchItem.LastSeqNum.Cmp(requestedLastSeqNum) >= 0 {
+						break
+					}
+				}
+			} else if comparison > 0 {
+				if startingIndex > 0 {
+					startingIndex--
+				}
+				for ; startingIndex > 0; startingIndex-- {
+					if q.broadcastMessages[startingIndex].FeedItem.BatchItem.LastSeqNum.Cmp(requestedLastSeqNum) >= 0 {
```

Comment:
I think this check needs to be changed to `<=`. Theoretically it can send an extra from that, but I don't think we need to worry about sending too many

Reply:
:+1: